{{ cookiecutter.project_slug }}/{{ cookiecutter.project_slug }}/apps/common/formatters.py

```diff
@@ -0,0 +1,25 @@
+from pythonjsonlogger import jsonlogger
+
+
+class DjangoRequestJsonFormatter(jsonlogger.JsonFormatter):
+    """
+    This class is used to serialize headers and body of the JSON Django request
+    into the `request` property in the record of `jsonlogger.JsonFormatter`.
+    """
+
+    def add_fields(self, log_record, record, message_dict):
+        if getattr(record, 'request', None):
+            def is_django_request(request):
+                return getattr(record.request, 'method', None)
+
+            def is_json_request(request):
+                return record.request.META.get('CONTENT_TYPE', None) == 'application/json'
+
+            if is_django_request(record.request) and is_json_request(record.request):
+                record.request = {
+                    'request': record.request,
+                    'headers': record.request.headers,
+                    'body': record.request.body.decode('UTF-8')
+                }
```

Comment:
Hmm, I think that the conditional should be applied to only the `body` property not the whole `request` object. The rationale behind this would be to log the headers no matter what (which may be crucial to debug other requests) and only oversized body is a problem with file upload.

Reply:
Yes, that's true, headers should go at all times, if present, and only body should be filtered on some conditions.